src/components/app/data/queries/utils.js

```diff
@@ -0,0 +1,37 @@
+import { matchPath } from 'react-router-dom';
+import { queryEnterpriseLearnerDashboardBFF } from './queries';
+import { isBFFEnabledForEnterpriseCustomer } from '../utils';
+
+/**
+ * Resolves the appropriate BFF query function to use for the current route.
+ * @param {string} pathname - The current route pathname.
+ * @returns {Function|null} The BFF query function to use for the current route, or null if no match is found.
+ */
+export function resolveBFFQuery(pathname, options = {}) {
+  // Define route patterns and their corresponding query functions
```

Comment:
nit: this comment technically refers to `routeToBFFQueryMap` below; should it be co-located?